### ISSUE TITLE
Bug: Allow to hold CTRL, SHIFT, ALT and WIN keys

### ISSUE THREAD

**Issue by imjinjia** [ayufan/steam-deck-tools#81]
Every windows key is turbo on any mode , such as back buttons and a button , b button...

**Comment by ayufan** [comment]
Impossible. Please re-open if you have more information on what and when.

**Comment by imjinjia** [comment]
> Impossible. Please re-open if you have more information on what and when.
@ayufan 

First , thank you for your work , and forgive me for my poor English . 

I've tried everything , re-open or re-install , still the same.

Please try two keys shortcut in setting , like ctrl + c , shortcut won't work easily because all the windows keys are turbo can't hold the keys .

Please test it with keyboardtestutility.
[keyboardtestutility.zip](https://github.com/ayufan/steam-deck-tools/files/10909040/keyboardtestutility.zip)

**Comment by ayyoubmiftah** [comment]
@imjinjia do you found a solution to this probleme ?

**Comment by imjinjia** [comment]
@ayyoubmiftah I didn't find a solution to this probleme.
The author does not believe that SteamController has this problem，i think you should create an issue.

**Comment by ayufan** [comment]
I need more details, so if someone would have or record a video with it, it would make it easier for me to see what and when it happens :)

**Comment by imjinjia** [comment]
@ayufan  hold any button on desktop mode , if you test it with keyboardtestutility , you will see the probleme.

**Comment by ayufan** [comment]
Sorry, I'm a little sceptical on running some random programs from internet. Can you describe different method of reproducing it?

**Comment by imjinjia** [comment]
Hold an button , like a button or b button , and it just keep repeat.

**Comment by ayufan** [comment]
Yes, but what I would expect to see?

**Comment by ayyoubmiftah** [comment]
@imjinjia @ayufan for example if you long click in Ybuton in steam deck to do long press in shift button for example but shift button (clicked down and up many times)

**Comment by ayufan** [comment]
It is expected that keys do generate repeats, this is how keyboard work. Some do should not like (SHIFT), but if you press Key-A it will generate first press and repeats.

So, maybe we should not generate repeats on function keys?

**Comment by ayyoubmiftah** [comment]
@ayufan yeah exactly , do you know how to do long press for 2 seconds in Key (for example :VK_A) with c#

**Comment by ayufan** [comment]
@ayyoubmiftah There's nothing such as long press with a correct interval. It is events generated:
- WK_KEYDOWN
- and repeats if needed
- WK_KEYUP

**Comment by imjinjia** [comment]
I just want hold ctrl key to multi-select.

**Comment by ayufan** [comment]
Will fix it, fixed description here.

**Comment by ayyoubmiftah** [comment]
@imjinjia <<<<I just want hold ctrl key to multi-select.>>>> 
Me too

Thanks @ayufan

**Comment by ayyoubmiftah** [comment]
@imjinjia I found a solution with https://github.com/AntiMicroX/antimicrox it can do a long press emulation and work Great with SDT

**Comment by imjinjia** [comment]
@ayyoubmiftah thank you for telling me.

**Comment by ayyoubmiftah** [comment]
@imjinjia problem is solved .
if you are familiar to c# work with this function :

[DllImport("user32.dll", SetLastError = true)]
static extern void keybd_event(byte bVk, byte bScan, int dwFlags, int dwExtraInfo);


public static void KeyPress(int keycode, int delay = 0)
{
   keybd_event((byte)keycode, 0x0, 0, 0);// presses
   System.Threading.Thread.Sleep(delay);
   keybd_event((byte)keycode, 0x0, 2, 0); //releases
}

or used like this 

if(c.CUSTOMBTN.justPressed())
keybd_event((byte)keycode, 0x0, 0, 0);
else if(c.CUSTOMBTN.pressed())
 keybd_event((byte)keycode, 0x0, 2, 0);

and finally you can make a long click 👍 🏆 (Now just waiting for custom butons/profiles with interface )

**Comment by ofbakirci** [comment]
> @imjinjia problem is solved . if you are familiar to c# work with this function :
> 
> [DllImport("user32.dll", SetLastError = true)] static extern void keybd_event(byte bVk, byte bScan, int dwFlags, int dwExtraInfo);
> 
> public static void KeyPress(int keycode, int delay = 0) { keybd_event((byte)keycode, 0x0, 0, 0);// presses System.Threading.Thread.Sleep(delay); keybd_event((byte)keycode, 0x0, 2, 0); //releases }
> 
> or used like this
> 
> if(c.CUSTOMBTN.justPressed()) keybd_event((byte)keycode, 0x0, 0, 0); else if(c.CUSTOMBTN.pressed()) keybd_event((byte)keycode, 0x0, 2, 0);
> 
> and finally you can make a long click 👍 🏆 (Now just waiting for custom butons/profiles with interface )

@ayyoubmiftah Would you mind briefly explaining or pointing in the right direction about how to get these C# functions work? I'm a tech savvy and familiar with coding but that's all
